src/dataflow/src/sink/kafka.rs

```diff
@@ -610,26 +602,16 @@ impl KafkaSinkState {
         Err(last_error)
     }
 
-    async fn begin_transaction(&self) -> KafkaResult<()> {
-        self.producer.begin_transaction().await
-    }
-
-    async fn commit_transaction(&self) -> KafkaResult<()> {
```

Comment:
I thought about changing this function (and similar ones) to just do `self.retry_on_txn_error(|p| p.commit_transaction()).await?;` but I think that could actually make it less clear that it's important to call inside `retry_on_txn_error` because it isn't so obvious at each invocation.

However, I'm very open to disagreement on this point and can make the change if anyone feels strongly